src/coreclr/jit/emit.h

```diff
@@ -1007,6 +1007,16 @@ class emitter
                 regNumber      _idReg4 : REGNUM_BITS;
             };
 
+            struct
```

Comment:
@dotnet/jit-contrib I'm not proud of doing this.

The idea is to use the space that stores `_idReg4` but use it for `insSvePrfop`.

I tried doing a union inside the struct above like so:
```cpp
            struct
            {
                // This 32-bit structure can pack with these unsigned bit fields
                emitLclVarAddr iiaLclVar;
                unsigned       _idRegBit : 1; // Reg3 is scaled by idOpSize bits
                GCtype         _idGCref2 : 2;
                regNumber      _idReg3 : REGNUM_BITS;
                union
                {
                    regNumber _idReg4 : REGNUM_BITS;
                    insSvePrfop _idSvePrfop : REGNUM_BITS;
                }
            };
```

But this fails a compile-time assert:
`    C_ASSERT(sizeof(emitter::instrDesc) == (SMALL_IDSC_SIZE + sizeof(void*)));`

This tells me that the size of the `instrDesc` changed when I introduced the nested union. Perhaps it is due to alignment? Is there an alternative way?

Reply:
IIUC, C++ is very picky about packing bitfields in that they all need to be of the same type. The "union" probably breaks that. You could have the `idSvePrfop` functions just use `_idReg4` directly, maybe with casting. Still ugly :-)

Reply:
From my understanding `instrDesc` must be 8 bytes. We sometimes allocate more space for it for Debug or if JitDisasm is enabled in order to store more info for every record +8 byte -- I guess you can go similar route or make a side dictionary? Presumably, unconditional extension for instrDesc from 8 bytes to 16 will be a major TP regression. Afair, it was already painful to increase bit width for `_idIns` field because of SVE

Reply:
I prefer the casting option over the version in the PR. You could even go as far as renaming `_idReg4` and making it unsigned instead of `regNumber` so that it's explicit it's used for multiple things.

If it's alignment causing the issues, then have you tried this way:
```
            struct
            {
                emitLclVarAddr iiaLclVar;
                union
                {
                    regNumber _idReg4 : REGNUM_BITS;
                    insSvePrfop _idSvePrfop : REGNUM_BITS;
                };
                unsigned       _idRegBit : 1; // Reg3 is scaled by idOpSize bits
                GCtype         _idGCref2 : 2;
                regNumber      _idReg3 : REGNUM_BITS;

            };
```
`emitLclVarAddr` is 32bits, so this will put  `_idReg4/_idSvePrfop` to 32bit alignment. Or if that doesn't work, try pushing the union to the top of the struct.


`// This 32-bit structure can pack with these unsigned bit fields` - This comment doesn't make sense as `emitLclVarAddr` is 32bits, making the full struct 47bits. Probably best to remove. It's left over from when the Arm32 and Arm64 code was entangled.


Whatever solution we end up with, in a later PR we should refactor `_idSvePattern` so that it uses the same mechanism.

Reply:
Yea, I'm definitely leaning towards doing that, it's not super great to have a bunch of dummy padding fields.